clowder-framework/clowder2
# 1147 clowder 2 helm chart clean up

This PR addresses below:

- Allow overwrite with existing secrets to different compoents e.g. backend, geoserver
- Update default values so that:
  - Allow ingress to be turned off e.g. rabbitmq, minio
  - Turn off geoserver by default
- Fix duplicated hostanme in TLS
- Use fullname instead of name in various places
- Allow overwrite of the env variables on backend and frontend
----

To test:

2. Backward compatibility with clowder2 demo instance
- switch context `kubectl config use-context software-dev `
- acquire the "chen-clowder2-software-dev.yaml" from the box folder https://uofi.app.box.com/folder/205332339557
- place it under deployment/charts/clowder2 folder
- run `helm diff upgrade --namespace clowder2 clowder2 . --values chen-clowder2-software-dev.yaml` this should have minimum diffs

-----
!!! Already deployed so don't repeat:

1. Work with littlebeats yaml file
see this PR: https://git.ncsa.illinois.edu/kubernetes/clusters/clowder/-/merge_requests/5
- set up connection to "clowder". 
- switch context `kubectl config use-context clowder `
- place the "littlebeats.yaml" in the deployment/charts/clowder2 folder
- run ` helm upgrade --namespace littlebeats littlebeats . --values littlebeats.yaml --allow-unreleased`  this should
